File: src/main/java/com/javarush/island/shevchenko/entities/animals/herbivores/Deer.java
package com.javarush.island.shevchenko.entities.animals.herbivores;

import com.javarush.island.shevchenko.entities.organism.Animal;
import com.javarush.island.shevchenko.entities.organism.interfaces.*;
import com.javarush.island.shevchenko.settings.IslandSettings;
import com.javarush.island.shevchenko.util.Randomizer;

/**
 * Класс Deer представляет существо лось на острове животных.
 * Лоси являются травоядными и используют поведение из интерфейса Herbivorous для питания.
 */
public class Deer extends Animal implements Herbivorous {

    public Deer() {
        this.aClass = Deer.class;
        this.weight = Randomizer.getRandom(IslandSettings.ANIMAL_PARAMETERS.get(Deer.class)[0] / 2, IslandSettings.ANIMAL_PARAMETERS.get(Deer.class)[0]);
    }
}

File: src/main/kotlin/ru/cooper/island/core/model/animal/herbivores/Deer.kt
package ru.cooper.island.core.model.animal.herbivores

import ru.cooper.island.core.model.animal.util.initAnimal
import ru.cooper.island.core.model.Animal
import ru.cooper.island.core.model.traits.Herbivorous

/**
 * Олень (травоядное животное)
 * Deer (herbivorous animal)
 */
class Deer : Animal(), Herbivorous {
    init { initAnimal<Deer>() }
}
